### PR TITLE
Check if 404 template exists

### LINKED ISSUE #366: Not all themes have 404 templates
I'm seeing lots of these errors:

`PHP Fatal error:  Uncaught ValueError: Path cannot be empty in /home/blog/wp-content/plugins/webmention/templates/webmention-comment.php:10`

https://github.com/pfefferle/wordpress-webmention/blob/216b1b9727437a3c3d44d51476e4c551a14a9f6e/templates/webmention-comment.php#L10

That should probably be replaced with:

```php
if ( null !== get_query_template( '404' ) ) {
   include( get_query_template( '404' ) );
}
```

Which is similar to the approach in https://github.com/pfefferle/wordpress-webmention/blob/main/includes/class-http-gone.php

### PR DESCRIPTION
**Pull request by edent**
Fixes #366

### CODE REVIEW

File: templates/webmention-comment.php [hunk 1]
```diff
@@ -7,7 +7,12 @@
 if ( ! $comment ) {
 	status_header( 404 );
 	nocache_headers();
-	include get_query_template( '404' );
+	// check if theme has a 404.php template
+	$template_404 = get_query_template( 404 );
+	// return 404 template
+	if ( $template_404 ) {
+		include get_query_template( '404' );
```
Comment (pfefferle) [inline templates/webmention-comment.php:14]: I think we could re-use `$template_404` here, instead of querying it again?

**Comment by pfefferle** [comment]
Will that result in an empty page (White screen) if there is no template?

**Comment by edent** [comment]
Yes - which is no different from what happens now.

**Comment by pfefferle** [comment]
sure, but if we are already improving it... ;)

File: templates/webmention-comment.php [hunk 2]
```diff
@@ -7,7 +7,12 @@
 if ( ! $comment ) {
 	status_header( 404 );
 	nocache_headers();
-	include get_query_template( '404' );
+	// check if theme has a 404.php template
+	$template_404 = get_query_template( 404 );
+	// return 404 template
+	if ( $template_404 ) {
+		include $template_404; 
+	}
```
Comment (pfefferle) [inline templates/webmention-comment.php:15]: what do you think about:

```PHP
	if ( $template_404 ) {
		include $template_404;
		die();
	} else {
		wp_die(
			__( 'Page not found', 'webmention' ),
			__( '404', 'webmention' )
		);
	}
```

Comment (edent) [reply templates/webmention-comment.php:15]: I think that makes sense.

**Comment by dshanske** [comment]
Actually, the template hierarchy says that if there is no 404 page, it should return index.php.

**Comment by pfefferle** [comment]
Have I used the wrong method @dshanske ?

**Comment by dshanske** [comment]
> Have I used the wrong method @dshanske ?

I'm just asking, shouldn't it redirect to the index instead of dying as a white screen?

**Review by dshanske — APPROVED** [review]
Looks good.

**Pull request merged**